### PR TITLE
milestone6: kelwilson-portfolio;

### PR DESCRIPTION
**Pull request by kelwilson**
### **Validate Contact Form**

- Add a function to the form-validation.js file
- Add a regex expression to Check if the email matches the regex for email input
- Add an if statement to check if the email is lowercase
- Add an event listener to either display an error message when the input field requirements are not met or to submit when they are met

### CODE REVIEW

**Review by Kossi-stack — CHANGES_REQUESTED** [review]
## Hi @kelwilson,

**Good job so far! 👍🏻**
There are some issues that you still need to work on to go to the next project but you are almost there!💪

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Required Changes ♻️

 - [ ] Kindly edit the PR's title to something more descriptive. it can be "Validate contact form Project PR".
 - [ ] Please make sure to use [this readme template](https://github.com/microverseinc/readme-template/blob/master/README.md) and please be sure to edit all of the sections to better suit your project.
 Please fix your readme
Capturereme



 _Check the comments under the review for more changes_.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: form-validation.js [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-//       });
-//     }
-//   }
-// }
+function checkemail() {
+  const form = document.getElementById('contact-form');
+  const email = form.elements.email.value;
+  const re = /@/g;
+  const lower = email.toLowerCase();
+  // Check if email matches regex for email input
```
Comment (Ismailco) [inline form-validation.js]: - [x] [OPTIONAL] You don't need to comment on all your code, kindly if the code is self-explanatory consider removing the comments

File: form-validation.js [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+  if (re.test(String(email).toLowerCase()) === true) {
+    if (lower === email) { // Check if email is lowercase
+      form.submit(); // Submit if it's correct
+    } else {
+      form.addEventListener('submit', (event) => {
+        event.preventDefault(); // Prevent form submission
+        const error = document.createElement('aside'); // Create container for error message
+        error.textContent = 'Please make sure the email is all lower case'; // Error message
```
Comment (Ismailco) [inline form-validation.js]: - [x] [OPTIONAL] The error message displayed even after the use correct the email to use only lower case letters, kindly consider fixing this issue

File: form-validation.js [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
-//         }, 500);
-//         form.appendChild(error);
-//         setTimeout(() => { // Delay for function to remove appended children
-//           form.removeChild(document.getElementById('error-popup'));
-//         }, 3500);
-//       });
-//     }
-//   }
-// }
+function checkemail() {
```
Comment (Ismailco) [inline form-validation.js:1]: - [x] As the previous reviewer mentioned kindly fix your README so it looks more professional, The live demo header is repeating 3 times, and there are some issues with the markdown. You can checkout [the README template from microverse](https://github.com/microverseinc/readme-template/blob/master/README.md).
Screenshot_20220618_172251

File: form-validation.js [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+  const form = document.getElementById('contact-form');
```
Comment (Ismailco) [inline form-validation.js:2]: - [x] [OPTIONAL] The previous reviewer should invalidate this PR, sorry for this mistake but you should implement the above requirements in all repositories in your pair-programming group, kindly use peer-programming in your project

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @kelwilson,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Github Flow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by koachcode — APPROVED** [review]
Hi @kelwilson , 

# STATUS: APPROVED 🟢

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

well-1

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**